### ISSUE TITLE
Support reading configuration from stdin

### ISSUE THREAD

**Issue by howardjohn** [golangci/golangci-lint#3739]
### Your feature request related to a problem? Please describe.

We would like to read configuration from stdin. The reason is we want to deep merge a few configuration files with `yq` and pipe them in. This sort of works, but gives an error.

```
$ echo '' | golangci-lint run -c /dev/stdin
WARN [config_reader] Can't pretty print config file path: can't eval symlinks for path /dev/stdin: lstat /proc/688987/fd/pipe:[50935730]: no such file or directory
```

(in real world our command is not, echo, of course)

### Describe the solution you'd like.

Support config from stdin without warning/error

### Describe alternatives you've considered.

For our specific issue, passing multiple config files and having them deep merged would also be suitable. However, this may be complex as different folks have different meanings of "merge"

**Comment by ldez** [comment]
Hello,

It's not an error but a warning, the warning doesn't stop the analysis.

I guess `yq` is https://github.com/kislyuk/yq.

I don't know the possibilities of `yq` but you can already do: 

```bash
golangci-lint run --config <(yq '. *= load("bar-golangci.yml")' foo-golangci.yml)
```

We will not create a "merge" of configuration files because merging files will lead to unnecessary complexity.

**Comment by ldez** [comment]
I create a PR to remove the warning when the configuration file explicitly stdin.

#3740

**Comment by ldez** [comment]
But my fix doesn't work when the command is piped:

```console
$ yq '. *= load("bar-golangci.yml")' foo-golangci.yml | golangci-lint run --config /dev/stdin
WARN [config_reader] Can't pretty print config file path: can't eval symlinks for path /dev/stdin: lstat /proc/3741263/fd/pipe:[62208537]: no such file or directory 
```

I need to think of a better way to handle that.

**Comment by ldez** [comment]
I was not using my own fix :facepalm:

Everything works as expected.
